### DISCUSSION TITLE
fwupd -- Does this output of "fwupdtool security --force" look right ???

### DISCUSSION THREAD

**Discussion by Mrmad1966** [fwupd/fwupd#6075] [Q&A] [answered]
I'd appreciate someone more knowledgable than I to advise me please.. Pulling my hair out with a dual boot system. My Fedora is now so slow at boot time. It seems to point to fwupd as a problem. As per the tile I have provided the output below guidance on how to even start trouble shooting would be so useful please
sudo fwupdtool security --force
Loading…                 [*******************                    ]05:24:01.274 FuPluginIntelMe      failed to get public key using file-id 0x40002300, section fpf [0x8]: generic failure [0xd]
05:24:01.275 FuPluginIntelMe      failed to get public key using file-id 0x40002300, section uep [0x4]: generic failure [0xd]
05:24:01.279 FuPluginIntelMe      failed to get public key using file-id 0x40005b00, section fpf [0x8]: generic failure [0xd]
05:24:01.280 FuPluginIntelMe      failed to get public key using file-id 0x40005b00, section uep [0x4]: generic failure [0xd]
05:24:01.283 FuPluginIntelMe      failed to get public key using file-id 0x40005c00, section fpf [0x8]: generic failure [0xd]
05:24:01.284 FuPluginIntelMe      failed to get public key using file-id 0x40005c00, section uep [0x4]: generic failure [0xd]
Loading…                 [************************************** ]
Host Security ID: HSI:0! (v1.9.3)

HSI-1
✔ MEI key manifest:              Valid
✔ SPI write:                     Disabled
✔ Supported CPU:                 Valid
✔ TPM empty PCRs:                Valid
✔ TPM v2.0:                      Found
✔ UEFI bootservice variables:    Locked
✔ UEFI platform key:             Valid
✔ UEFI secure boot:              Enabled
✔ csme override:                 Locked
✔ csme v0:14.0.31.1120:          Valid
✘ Platform debugging:            Enabled
✘ SPI BIOS region:               Unlocked
✘ SPI lock:                      Disabled
✘ csme manufacturing mode:       Unlocked

HSI-2
✔ Intel BootGuard:               Enabled
✔ Platform debugging:            Locked
✔ TPM PCR0 reconstruction:       Valid
✘ IOMMU:                         Not found
✘ Intel BootGuard ACM protected: Invalid
✘ Intel BootGuard OTP fuse:      Invalid
✘ Intel BootGuard verified boot: Invalid

HSI-3
✘ Intel BootGuard error policy:  Invalid
✘ Intel CET Enabled:             Not supported
✘ Pre-boot DMA protection:       Invalid
✘ Suspend-to-idle:               Disabled
✘ Suspend-to-ram:                Enabled

HSI-4
✔ Intel SMAP:                    Enabled
✘ Encrypted RAM:                 Not supported

Runtime Suffix -!
✔ Linux kernel lockdown:         Enabled
✔ Linux swap:                    Encrypted
✔ fwupd plugins:                 Untainted
✘ Linux kernel:                  Tainted

This system has a low HSI security level.
 » https://fwupd.github.io/hsi.html#low-security-level

This system has HSI runtime issues.
 » https://fwupd.github.io/hsi.html#hsi-runtime-suffix

**Comment by hughsie** [comment]
Why do you think fwupd is the cause of the slow boot problem? How long does `sudo fwupdtool get-devices` take to complete?

**Reply by Mrmad1966** [reply]
I'm not saying I think it is for sure as I'm not knowledgable enough to say that. All I know for sure is my original Fedora install (dual booted with Win10) was rapid at booting I mean around the 10 to 15 second mark but now its a minute and a half. looking at the results of systemd-analyze seems to point at the main culprits are (loader around 13secs) & (kernel around 41secs).

In reply to your question,  sudo fwupdtool get-devices took 38 seconds to complete..

**Reply by hughsie** [reply]
>  sudo fwupdtool get-devices took 38 seconds to complete

Okay that's waaaay too long. Can you attach the output of ` sudo fwupdtool get-devices -vv` please.

**Comment by Mrmad1966** [comment]
[`]([fwupdtool get-devices -vv.txt](https://github.com/fwupd/fwupd/files/12320048/fwupdtool.get-devices.-vv.txt))

**Reply by hughsie** [reply, marked as answer]
There's something wrong with your TPM:

                ../src/fu-engine.c:7746:/sys/devices/platform/IFX0785:00/tpm/tpm0 [27711.19ms]:
                    ../src/fu-engine.c:7713:query-possible-plugins [27711.10ms]:
                        ../libfwupdplugin/fu-plugin.c:1536:tpm [27711.09ms]:
                            :open [27698.24ms]

Is it disabled in the system firmware? does `sudo tpm2_pcrread` also take a very long time?

**Reply by Mrmad1966** [reply]
I love you.......It was enabled, setting to disabled has shaved a minute off. Thank you so much for your expertise and help

**Comment by Mrmad1966** [comment]
Disabled TPM
